src/lib/icon/icon-registry.ts

```diff
@@ -48,18 +48,39 @@ export function getMatIconNoHttpProviderError(): Error {
  * @param url URL that was attempted to be sanitized.
  * @docs-private
  */
-export function getMatIconFailedToSanitizeError(url: SafeResourceUrl): Error {
+export function getMatIconFailedToSanitizeUrlError(url: SafeResourceUrl): Error {
   return Error(`The URL provided to MatIconRegistry was not trusted as a resource URL ` +
                `via Angular's DomSanitizer. Attempted URL was "${url}".`);
 }
 
+/**
+ * Returns an exception to be thrown when a HTML string couldn't be sanitized.
+ * @param template HTML that was attempted to be sanitized.
+ * @docs-private
+ */
+export function getMatIconFailedToSanitizeTemplateError(template: SafeHtml): Error {
+  return Error(`The template provided to MatIconRegistry was not trusted as safe HTML by ` +
+               `Angular's DomSanitizer. Attempted template was "${template}".`);
+}
+
+
 /**
  * Configuration for an icon, including the URL and possibly the cached SVG element.
  * @docs-private
  */
 class SvgIconConfig {
-  svgElement: SVGElement | null = null;
-  constructor(public url: SafeResourceUrl) { }
+  url: SafeResourceUrl | null;
+  svgElement: SVGElement | null;
+
+  constructor(url: SafeResourceUrl);
+  constructor(svgElement: SVGElement);
+  constructor(data: SafeResourceUrl | SVGElement) {
+    if (data instanceof SVGElement) {
```

Comment:
Will referencing `SVGElement` cause any issues on platform-server?

Reply:
Yep, looks like it does. I've switched it to a hackier approach that should work everywhere.